cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java

```diff
@@ -64,169 +65,432 @@ public List<Parameter> getParameters() {
+        // Update dependencies
+        updateDependenciesIGM(subsetModels, context.getBoundaryEqId(), context.getBoundaryTpId());
+
+        // Export requested subsets
+        context.setExportEquipment(requestedSubsets.contains(CgmesSubset.EQUIPMENT));
+        String baseName = getBaseName(context, dataSource, network);
+        for (CgmesSubset subset : requestedSubsets) {
+            String fileName = baseName + "_" + subset.getIdentifier() + ".xml";
+            subsetExport(network, subset, fileName, dataSource, context, subsetModels.get(subset));
         }
 
-        try {
-            List<String> profiles = Parameter.readStringList(getFormat(), params, PROFILES_PARAMETER, defaultValueConfig);
-            checkConsistency(profiles, network, context);
-            if (profiles.contains("EQ")) {
-                try (OutputStream out = new BufferedOutputStream(ds.newOutputStream(filenameEq, false))) {
-                    XMLStreamWriter writer = XmlUtil.initializeWriter(true, INDENT, out);
-                    EquipmentExport.write(network, writer, context);
-                }
-            } else {
-                addSubsetIdentifiers(network, "EQ", context.getExportedEQModel());
-                context.getExportedEQModel().setId(context.getNamingStrategy().getCgmesId(network));
-            }
-            if (profiles.contains("TP")) {
-                try (OutputStream out = new BufferedOutputStream(ds.newOutputStream(filenameTp, false))) {
-                    XMLStreamWriter writer = XmlUtil.initializeWriter(true, INDENT, out);
-                    TopologyExport.write(network, writer, context);
-                }
-            } else {
-                addSubsetIdentifiers(network, "TP", context.getExportedTPModel());
+        context.getNamingStrategy().debug(baseName, dataSource);
+    }
+
+    /**
+     * Initialize the model (= the metadata information) that is used by the export.
+     * If existing, the network model extension is used for the initialization.
+     * If existing, optional parameters are also used for the initialization.
+     * If both are present, the optional parameters prevail the values in the network extension.
+     * @param network The network in which to look for an existing model extension as basis for initialization.
+     * @param subset The subset of the model to initialize.
+     * @param context The context used by the export.
+     * @param mainNetwork A boolean indicating whether the exported network is the main network (not a subnetwork).
+     * @param modelUpdate A boolean indicating whether the model version and ID should be updated.
+     * @return A model with all necessary metadata information for the export.
+     */
+    public static CgmesMetadataModel initializeModelForExport(
+            Network network, CgmesSubset subset, CgmesExportContext context, Boolean mainNetwork, Boolean modelUpdate) {
```

Comment:
why not use primitive `boolean`'s here? the conditions below will be more concise and readable